tap_polarsh/streams.py

```diff
@@ -89,3 +89,151 @@ def get_url_params(
             **super().get_url_params(context, next_page_token),
             "organization_id": context["organization_id"] if context else None,
         }
+
+
+class _BaseBenefits(PolarStream):
+    """Base benefits stream."""
+
+    path = "/api/v1/benefits"
+    primary_keys = ("id",)
+
+    parent_stream_type = Organizations
+
+    benefit_type: str
+
+    def get_url_params(
+        self,
+        context: Context | None,
+        next_page_token: int | None,
+    ) -> dict[str, t.Any]:
+        """Get URL query parameters.
+
+        Args:
+            context: The context of the stream.
+            next_page_token: The next page token.
+
+        Returns:
+            The URL query parameters.
+        """
+        return {
+            **super().get_url_params(context, next_page_token),
+            "organization_id": context["organization_id"] if context else None,
+            "type": self.benefit_type,
+        }
+
+    def get_child_context(  # noqa: PLR6301
+        self,
+        record: dict[str, t.Any],
+        context: Context | None,  # noqa: ARG002
+    ) -> dict[str, t.Any]:
+        """Get child context.
+
+        Args:
+            record: The record.
+            context: The context of the stream.
+
+        Returns:
+            The child context.
+        """
+        return {
+            "benefit_id": record["id"],
+        }
+
+
+class _BaseBenefitsGrants(PolarStream):
+    """Base benefits grants stream."""
+
+    path = "/api/v1/benefits/{benefit_id}/grants"
+    primary_keys = ("id",)
+
+    swagger_ref: str = "BenefitGrant"
+
+
+class BenefitsCustom(_BaseBenefits):
+    """Custom benefits stream."""
+
+    name = "benefits_custom"
+    benefit_type = "custom"
+    swagger_ref: str = "BenefitCustom"
```

Comment:
Different benefits streams do have different schemas so it makes sense to split them, but benefit grants have a unique schema.